build/rocm/dev_build_rocm.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2024 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE(ruturaj4): This script automates the build process for JAX and XLA on ROCm, 
+# allowing for optional uninstallation of existing packages, and custom paths for ROCm and XLA repositories.
+
+import os
+import sys
+import shutil
+import argparse
+import subprocess
+
+def get_rocm_version():
+    try:
+        version = subprocess.check_output("cat /opt/rocm/.info/version | cut -d '-' -f 1", shell=True)
+        return version.decode("utf-8").strip()
+    except subprocess.CalledProcessError as e:
+        print(f"Error fetching ROCm version: {e}")
+        return None
+
+
+def get_rocm_target():
+    try:
+        target_info = subprocess.check_output("rocminfo | grep gfx | head -n 1", shell=True)
+        target = target_info.decode("utf-8").split()[1]
+        return target
+    except subprocess.CalledProcessError as e:
+        print(f"Error fetching ROCm target: {e}")
+        return None
+
+
+def uninstall_existing_packages(packages):
+    for pkg in packages:
+        try:
+            subprocess.run(f"python3 -m pip uninstall {pkg} -y", shell=True, check=True)
+            print(f"Successfully uninstalled {pkg}")
+        except subprocess.CalledProcessError as e:
+            print(f"Failed to uninstall {pkg}: {e}")
```

Comment:
You can avoid a shell injection issue, and I think simplify a bit:
```suggestion
      cmd = ["python3", "-m", "pip", "uninstall", "-y"]
      cmd.extend(packages)
      
      try:
          subprocess.run(cmd, check=True)
          print(f"Successfully uninstalled {packages}")
      except subprocess.CalledProcessError as e:
          print(f"Failed to uninstall {packages}: {e}")
```